src/store/modules/game.js

```diff
@@ -168,6 +172,8 @@ export default {
       if (Object.hasOwnProperty.call(newGame, 'p1Ready')) state.p1Ready = newGame.p1Ready;
       if (Object.hasOwnProperty.call(newGame, 'passes')) state.passes = newGame.passes;
       if (Object.hasOwnProperty.call(newGame, 'players')) state.players = cloneDeep(newGame.players);
+      if (Object.hasOwnProperty.call(newGame, 'spectatingUsers'))
```

Comment:
@itsalaidbacklife less about this PR and more higher order, do we always need has own property or would some other check here be better long term?

Reply:
I think typescript would save us the need altogether. It's basically a paranoid type safety check that we could skip if we had proper types on our client and server

Reply:
The other thing this does is allow for partial updates. Which granted we could do with a spread instead